### ISSUE TITLE
[Feature Request] add scrollElement to useWindowScroll

### ISSUE THREAD

**Issue by acapro** [petyosi/react-virtuoso#320]
This is by far one of the best windowing tools I have tried, and would love to be able to use it, but there is one feature that is sadly still missing and I have not yet been able to find a work around this yet.

I have a more or less complex layout with multiple overflowing divs and scrollbars, and it would be an awesome feature if the useWindowScroll could be extended to support other wrapping scrollable (overflow) elements besides the window. 

In react-virtualized there is a WindowScroller component that does just this, it takes an html element parameter called scrollElement (https://github.com/bvaughn/react-virtualized/blob/master/docs/WindowScroller.md)

Here is what I am trying to do:
<img width="815" alt="Screenshot 2021-03-24 at 16 11 00" src="https://user-images.githubusercontent.com/4762057/112334187-8995c900-8cbb-11eb-916f-52553813ef37.png">

**Comment by petyosi** [comment]
Hey, 

What you explain is fairly simple to achieve. You can take a look at the commit which introduced the window scrolling behavior and extend it further. I am happy to accept a PR for that.

**Comment by maxhelsel** [comment]
I second this. My _only_ issue with Virtuoso is the fact it doesn’t have support for assigning a custom parent scroll element like React Virtualized WindowScroller.

My use case is within modals and containers that have other elements within them; RV WindowScroller makes it easy and convenient to inject a scrollElement ref and carry on. 

I forked this package and spent a few hours with it, but I’m not experienced whatsoever with packages and just couldn’t get a handle on everything going on within the src file. Unfortunately, I don’t have the time to familiarize myself enough to see it through. 

If @petyosi or anyone else capable of implementing this feature stumbles upon this, here’s the link to where the window scrolling feature was introduced to Virtuoso: [https://github.com/petyosi/react-virtuoso/compare/v1.2.7...v1.3.0](https://github.com/petyosi/react-virtuoso/compare/v1.2.7...v1.3.0). 

My thinking was to provide an optional scrollElement prop for Virtuoso component, to be swapped within buildWindowScroller if the scrollElement props exists.

- Uses window (via buildWindowScroller within src/List)
```
<Virtuoso 
  useWindowScroll={true} 
/>
```

- Uses custom scrollElement (via buildWindowScroller within src/List - swaps the customRef for the window). Figured it would be easiest just to try to slip the ref within the buildWindowScroller function and recycle all that logic.
```
<Virtuoso 
  useWindowScroll={true}
  scrollElement={myCustomRef} 
/>
```

Then below is the function to build the Scroller element when useWindowScoll={true} - this code is copied and unchanged from src/List. I think here would be the place to substitute the scrollElement for the window if the scrollElement prop exists. You'd also need to adjust the useScrollTop and the viewport functions too, I believe (maybe some others?)
```
buildWindowScroller({ usePublisher, useEmitter, useEmitterValue }: Hooks) {
  const Scroller: Components['Scroller'] = React.memo(function VirtuosoWindowScroller({ style, children, ...props }) {
    const scrollTopCallback = usePublisher('windowScrollTop')
    const ScrollerComponent = useEmitterValue('ScrollerComponent')!
    const smoothScrollTargetReached = usePublisher('smoothScrollTargetReached')
    const totalListHeight = useEmitterValue('totalListHeight')
    const { scrollerRef, scrollByCallback, scrollToCallback } = useScrollTop(
      scrollTopCallback,
      smoothScrollTargetReached,
      ScrollerComponent
    )

    useIsomorphicLayoutEffect(() => {
      scrollerRef.current = window
      return () => {
        scrollerRef.current = null
      }
    }, [scrollerRef])

    useEmitter('windowScrollTo', scrollToCallback)
    useEmitter('scrollBy', scrollByCallback)
    return createElement(
      ScrollerComponent,
      {
        style: { position: 'relative', ...style, ...(totalListHeight !== 0 ? { height: totalListHeight } : {}) },
        ...props,
      },
      children
    )
  })
  return Scroller
}
```
Not sure if this is even a feasible approach, but seemed less invasive than building out an entirely new buildScroller function for the custom scrollElement ref. 

So, that's where my heads at. Like I said, I don't have the experience with packages or time to spend to try to get this working, but I figured I could help contribute here to help see this feature realized. It's the only thing Virtuoso lacks compared to React Virtualized, and prevents it (at least for me) from being the complete windowing solution

**Comment by petyosi** [comment]
I'm afraid I'm very short on time recently. Also, it's hard to prioritize this since it is not related to the use cases of my day job.

**Comment by Guiw5** [comment]
we need some philanthropist people on this, I can't tag bill gates hahaha

**Comment by bhargavshah** [comment]
This is much needed. If possible please prioritize!

**Comment by petyosi** [comment]
@bhargavshah are you willing to sponsor the development of the feature?

**Comment by cvanem** [comment]
@petyosi Do you still need a PR for this?  I have something working but I noticed you have been working on a custom scroller example recently?

**Comment by petyosi** [comment]
Yes, a PR would be welcome.

**Comment by petyosi** [comment]
@maxhelsel @Guiw5 @acapro - please take a look at the #556 from @cvanem. The examples look right to me.

**Comment by petyosi** [comment]
Implemented in #556
